ceno_zkvm/src/e2e.rs

```diff
@@ -574,8 +575,18 @@ fn debug_memory_ranges(vm: &VMState, mem_final: &[MemFinalRecord]) {
         format_segments(vm.platform(), handled_addrs.iter().copied())
     );
 
-    for addr in &accessed_addrs {
-        assert!(handled_addrs.contains(addr), "unhandled addr: {:?}", addr);
+    let unhandled: BTreeSet<_> = accessed_addrs
+        .iter()
+        .filter(|addr| !handled_addrs.contains(addr))
+        .collect();
+
+    // TODO(Matthias): this should be an assert, but it's currently broken, because our caller doesn't handle hints.
```

Comment:
may you elaborate what the `caller` means in this context

Reply:
The function `emulate_program`, the only place that calls `debug_memory_ranges`.

Reply:
Do you want me to make that explicit in the comment in the code, too?